src/service/fixed-layer.js

```diff
@@ -557,9 +536,18 @@ export class FixedLayer {
 
     // Update `top`. This is necessary to adjust position to the viewer's
     // paddingTop.
-    if (state.top && (state.fixed || (state.sticky && !this.transfer_))) {
-      setStyle(element, 'top', `calc(${state.top} + ${this.paddingTop_}px)`);
+    // TODO
+    if (state.top) {
+      if (state.fixed || !this.transfer_) {
+        setStyle(element, 'top', `calc(${state.top} + ${this.paddingTop_}px)`);
+      } else if (this.committedPaddingTop_ === this.paddingTop_) {
+        setStyle(element, 'top', state.top);
```

Comment:
To offset for iOS's `padding-top`. Without these lines, we get fixed and stickies that remain `54px` off the top of the viewport. This case is explicitly for when the header is showing (in which case, adding 54px would be compounding the 54px `padding-top`). The next case is for when the header is hidden, in which case we need to subtract that 54px.